### ISSUE TITLE
Caught ticking v2

### ISSUE THREAD

**Issue by b0nkers96** [seymourimadeit/Piglin-Proliferation#32]
[crash-2024-09-18_23.49.17-server.txt](https://github.com/user-attachments/files/17055404/crash-2024-09-18_23.49.17-server.txt)

**Comment by seymourimadeit** [comment]
Yeah from what I can tell on initial glance this is probably just a weird thing with your Java. I’ll have to take a closer look but you may want to consider reinstalling Java 21

**Comment by seymourimadeit** [comment]
Actually you might just need to install the Java 21 JDK instead of completely reinstalling the Java JRE

**Comment by b0nkers96** [comment]
What’s the difference if i reinstall java 21 jdk and java jre? Is some of the files of java jdk corrupted?

**Comment by seymourimadeit** [comment]
The reason I said to either install the Java jdk or reinstall the jre is because the error being given refers to a missing random number generation algorithm, which should not possible considering it is embedded in Java by default. That being combined with the fact I can’t recreate it at all (nor on the modpack being played) leads me to suspect something is up with your files

**Comment by b0nkers96** [comment]
So basically uninstall my java files and reinstall fresh ones? How about my minecraft client should I do anything with it as i just started to use curseforge for modded minecraft and still brand new and learning how to use them. So i feel like i have probably did something wrong when installing mods, curseforge and my mc client w/ shaders

**Comment by seymourimadeit** [comment]
> So basically uninstall my java files and reinstall fresh ones?

I'd try installing the Java 21 JDK first and if that doesn't work then delete and reinstall Java.

**Comment by b0nkers96** [comment]
![IMG_1120](https://github.com/user-attachments/assets/e613cbe0-424f-424e-ace3-3d59ed4afc4e)
These are the java’s i have installed currently, the one you’re telling me i dont think i have it installed so that could be the issue? But again the minecraft im running it on is on a hosted server.

**Comment by seymourimadeit** [comment]
With this new information I'm pretty certain you've ought to install Java 21.

**Comment by b0nkers96** [comment]
![javajdk](https://github.com/user-attachments/assets/ebc4e55f-0e12-4728-8c55-6e85d7ecef45)
will this version be good to download?

**Comment by seymourimadeit** [comment]
That should be the right one

**Comment by b0nkers96** [comment]
![Screenshot_1](https://github.com/user-attachments/assets/3632cbd4-d677-481e-89ef-043bc65f4519)
ops i meant the windows one but x64 installer  or x64 msi installer

**Comment by seymourimadeit** [comment]
Either one should work

**Comment by b0nkers96** [comment]
![Screenshot_2](https://github.com/user-attachments/assets/f3a6e8bd-5cb7-40e6-8350-a6dbbb360a36)
apparently i already had it installed? which is the previous photo i attached earlier

**Comment by seymourimadeit** [comment]
Oh I thought you were installing the JRE, I'd say reinstall the JDK and then install the JRE just to make sure

**Comment by b0nkers96** [comment]
So install java jre? Is it called java jre or is there another name for it?

**Comment by b0nkers96** [comment]
![Screenshot_3](https://github.com/user-attachments/assets/d7ae85c5-bc16-4bf2-b02b-8f0313e5f2d6)
ah nvm i found it so apparently my jre is outdated?

**Comment by seymourimadeit** [comment]
My mistake, the JRE is apparently only for Java 8. Reinstalling the JDK should be fine.

**Comment by b0nkers96** [comment]
java.lang.IllegalArgumentException: No implementation of the random number generator algorithm "L32X64MixRandom" is available
	at java.base/java.util.random.RandomGeneratorFactory.findProvider(Unknown Source)
	at java.base/java.util.random.RandomGeneratorFactory.of(Unknown Source)
	at java.base/java.util.random.RandomGenerator.of(Unknown Source)
	at java.base/java.util.random.RandomGenerator.getDefault(Unknown Source)
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.common.attribute.RangedRandomAttributeModifierHolder$Instance.random(RangedRandomAttributeModifierHolder.java:120)
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.common.attribute.RangedRandomAttributeModifierHolder$Instance.randomIntAmount(RangedRandomAttributeModifierHolder.java:115)
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.common.items.BucklerItem.bucklerBash(BucklerItem.java:122)
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.PPEvents.onLivingTick(PPEvents.java:151)
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.SubscribeEventListener.invoke(SubscribeEventListener.java:49)
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:350)
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:315)
	at TRANSFORMER/neoforge@21.1.55/net.neoforged.neoforge.event.EventHooks.fireEntityTickPre(EventHooks.java:935)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerLevel.tickNonPassenger(ServerLevel.java:773)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.Level.mixinextras$bridge$accept$173(Level.java)
	at TRANSFORMER/neruina@2.1.2/com.bawnorton.neruina.handler.TickHandler.safelyTickEntities(TickHandler.java:92)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.Level.wrapOperation$bac000$neruina$catchTickingEntities$notTheCauseOfTickLag(Level.java:6318)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.Level.guardEntityTick(Level.java:570)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerLevel.lambda$tick$2(ServerLevel.java:420)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.entity.EntityTickList.forEach(EntityTickList.java:54)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:400)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1037)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:317)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:917)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:707)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:267)
	at java.base/java.lang.Thread.run(Unknown Source)

I am still getting it on my server hosted

**Comment by seymourimadeit** [comment]
Yeah dog this is a strange one. Did you make sure the server is also running on Java 21?

I do feel like this is starting to get out of my expertise (a java environment not finding a value it's supposed to have that I cannot recreate on my device), so I'd try going here to see what you could do to having your Java files find the RNG algorithm https://discord.gg/moddedmc

**Comment by b0nkers96** [comment]
![Screenshot_4](https://github.com/user-attachments/assets/ca3452e6-a2bb-42f4-a220-20518ebaef22)
![Screenshot_5](https://github.com/user-attachments/assets/75f2c629-c454-490b-a25e-e729b614dd66)
![Screenshot_6](https://github.com/user-attachments/assets/bf51a95f-3bc1-42c7-93e8-f90b96293cf3)

So i have checked the server is on java 21 and neoforge version is on 21.1.55 and the loader on my curse forge is on 21.1.54 but i have tested it on a single world and i do not get crashes? 
The modpack i have installed is BMC5. I have another thing I see on the log talking about neruina.

**Comment by b0nkers96** [comment]
[20Sep2024 04:56:17.956] [Server thread/WARN] [neruina/]: Neruina caught an exception, see below for cause
java.lang.IllegalArgumentException: No implementation of the random number generator algorithm "L32X64MixRandom" is available
	at java.base/java.util.random.RandomGeneratorFactory.findProvider(Unknown Source) ~[?:?]
	at java.base/java.util.random.RandomGeneratorFactory.of(Unknown Source) ~[?:?]
	at java.base/java.util.random.RandomGenerator.of(Unknown Source) ~[?:?]
	at java.base/java.util.random.RandomGenerator.getDefault(Unknown Source) ~[?:?]
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.common.attribute.RangedRandomAttributeModifierHolder$Instance.random(RangedRandomAttributeModifierHolder.java:120) ~[piglinproliferation-1.21-2.0.7.jar%23377!/:?]
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.common.attribute.RangedRandomAttributeModifierHolder$Instance.randomIntAmount(RangedRandomAttributeModifierHolder.java:115) ~[piglinproliferation-1.21-2.0.7.jar%23377!/:?]
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.common.items.BucklerItem.bucklerBash(BucklerItem.java:122) ~[piglinproliferation-1.21-2.0.7.jar%23377!/:?]
	at TRANSFORMER/piglinproliferation@2.0.7/tallestred.piglinproliferation.PPEvents.onLivingTick(PPEvents.java:151) ~[piglinproliferation-1.21-2.0.7.jar%23377!/:?]
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.SubscribeEventListener.invoke(SubscribeEventListener.java:49) ~[bus-8.0.2.jar%2367!/:?]
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:350) ~[bus-8.0.2.jar%2367!/:?]
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:315) ~[bus-8.0.2.jar%2367!/:?]
	at TRANSFORMER/neoforge@21.1.55/net.neoforged.neoforge.event.EventHooks.fireEntityTickPre(EventHooks.java:935) ~[neoforge-21.1.55-universal.jar%23270!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerLevel.tickNonPassenger(ServerLevel.java:773) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.Level.mixinextras$bridge$accept$173(Level.java) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/neruina@2.1.2/com.bawnorton.neruina.handler.TickHandler.safelyTickEntities(TickHandler.java:92) ~[Neruina-2.1.2-neoforge+1.21.jar%23368!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.Level.wrapOperation$bac000$neruina$catchTickingEntities$notTheCauseOfTickLag(Level.java:6318) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.Level.guardEntityTick(Level.java:570) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerLevel.lambda$tick$2(ServerLevel.java:420) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.entity.EntityTickList.forEach(EntityTickList.java:54) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:400) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1037) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:317) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:917) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:707) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:267) ~[server-1.21.1-20240808.144430-srg.jar%23269!/:?]
	at java.base/java.lang.Thread.run(Unknown Source) [?:?]

**Comment by seymourimadeit** [comment]
> So i have checked the server is on java 21 and neoforge version is on 21.1.55 and the loader on my curse forge is on 21.1.54 but i have tested it on a single world and i do not get crashes?

This is even weirder considering the fact I just tested it out on a server just now with no crashes in sight.

I made an uncommitted change that changes the RNG algorithm to an old style one, could you see if it works on your system? 

[testrandom.zip](https://github.com/user-attachments/files/17070099/testrandom.zip)

**Comment by b0nkers96** [comment]
I'll test it out, is it the same file version still just coded differently or?

**Comment by b0nkers96** [comment]
> > So i have checked the server is on java 21 and neoforge version is on 21.1.55 and the loader on my curse forge is on 21.1.54 but i have tested it on a single world and i do not get crashes?
> 
> This is even weirder considering the fact I just tested it out on a server just now with no crashes in sight.
> 
> I made an uncommitted change that changes the RNG algorithm to an old style one, could you see if it works on your system? (the .zip file has a .jar file with the fix, you just need to install it your minecraft instance)
> 
> If it works then I will probably test it further and release it to cf and modrinth by (hopefully) this weekend
> 
> [testrandom.zip](https://github.com/user-attachments/files/17070099/testrandom.zip)

What server tick rate do you use/ i have neruina on 500 ticks 
![Screenshot_7](https://github.com/user-attachments/assets/c947d9bd-7d54-4687-acc7-64566e941085)
![Screenshot_8](https://github.com/user-attachments/assets/6fba7396-5ca2-45a8-a8a2-4d5bf1733d76)

**Comment by seymourimadeit** [comment]
> I'll test it out, is it the same file version still just coded differently or?

Yeah

The server I did was just a pure run of Piglin Proliferation, nothing else

**Comment by b0nkers96** [comment]
> > I'll test it out, is it the same file version still just coded differently or?
> 
> Yeah
> 
> The server I did was just a pure run of Piglin Proliferation, nothing else

I tested it out the current version i am using from bmc5 on a single player world it was working fine, just on the server that i do not know the issue of.

**Comment by seymourimadeit** [comment]
> I tested it out the current version i am using from bmc5 on a single player world it was working fine, just on the server that i do not know the issue of.

Judging from how the logs only implicated Piglin Proliferation and Neurina (which shouldn't be a deciding factor in this crash) I can only assume that a pure run of Piglin Proliferation on a server would be fine. 

The .jar in the .zip file only slightly changes the code to not use the new RNG system java has and instead use the old one. Simply find your curseforge instance's option to open it's directory and drop the jar file into the mods folder that will replace the old one (do this on the server too)

**Comment by b0nkers96** [comment]
have not gotten a chance yet, will do it today though. I'll let you know. Was debating that it could be the server i am just renting rn. Server.pro do u know any good server host

**Comment by b0nkers96** [comment]
DOes having neoforge version matter i just check my neoforge version is 21.1.54 and the server is 21.1.55 and the latest neoforge version is 21.1.57

**Comment by seymourimadeit** [comment]
It shouldn't matter too much in this case

**Comment by seymourimadeit** [comment]
Just checking in since it's been a week, any luck on this?

**Comment by emerl66** [comment]
Never really had to actually use github before so i have no clue how to navigate any of this, it will take me a bit to find the file you suggested for me to use. Will update once I learn how to find anything.

**Comment by seymourimadeit** [comment]
https://github.com/seymourimadeit/Piglin-Proliferation/issues/32#issuecomment-2362838987

This is the file I am referring to in this thread.

**Comment by emerl66** [comment]
> [#32 (comment)](https://github.com/seymourimadeit/Piglin-Proliferation/issues/32#issuecomment-2362838987)
> 
> This is the file I am referring to in this thread.

Yes thank you so much! After having to do some moving around and finally being able to get to re-enabling the mod. I did what you asked and replaced the jar file with the file from the zip file you posted before. After launching up and looking for some brutes to test on I was able to swing at them during the dash animation and let multiple perform the animation without a single crash so far. Thank you so much!

Sorry for having to trouble you with something so simple.
